### DISCUSSION TITLE
Problem with unwanted connection to Google Fonts

### DISCUSSION THREAD

**Discussion by Banuk1616** [templaza/astroid-framework#601] [Q&A]
Hello, I'm currently working on a new website locally. I installed fonts locally and set everything to Local under Typography. Nevertheless, a connection to Google Fonts is established and the Arizona font is loaded. Maybe someone has a tip for me.
No extensions are installed yet.

**Comment by Chacapamac** [comment]
@Banuk1616 — I will try to help.

First this is mini tutorial “**How to use Local Font in Astroid**”

**Use local fonts in Astroid “Typography” in the template manager**
‼️_Fonts Need to be loose in the “fonts” folder, cannot be in a subfolder_

**1-** Close the Astroid template manager. That is important because astroid initializes the local fonts on start up.

**2-** Download your fonts from a well know provider (google fonts, 1001fonts or other…) in the format otf or ttf or the more modern & performant woff format. — _(woff2 seem to not be compatible with Astroid yet…❓)_ 

**3-** Lets say the fonts file name is "oswald-v36-latin-regular.woff".

**4-** Login with your ftp client and navigate to the templates folder. — e.g. JOOMLAROOT/media/templates/site/your_template
Here open or create a folder name fonts "**fonts**" — e.g. JOOMLAROOT/media/templates/site/your_template/fonts

**5-** Copy your downloaded font file — e.g. "oswald-v36-latin-regular.woff" inside the newly created fonts folder.
_**NOTE:** need to be at the first level in the “fonts” folder to work_
— e.g. JOOMLAROOT/media/templates/site/your_template/fonts/oswald-v36-latin-regular.woff
‼️_NOT in a subfolder_
— e.g. JOOMLAROOT/media/templates/site/your_template/fonts/SUBFOLDER/oswald-v36-latin-regular.woff

**6-** Clear your whole Joomla cache for safety and reopen the astroid template options again.

**7-** Go to typography and you should find your custom font with a similar name under the “Custom Fonts” category drop down just after the “**System Fonts**” and before the “**Google fonts**” category
_**NOTE:** the final name of the font is given by the code inside the font file (at least with “woff” files) and can be different from the name of the file itself._

In this example (see image below) —> e.g. "oswald-v36-latin-regular.woff" become “Oswald Regular” in Astroid Typography settings
![0AED1CF8-625F-4758-8798-6E1BF8E74C6E](https://github.com/templaza/astroid-framework/assets/1907276/965abd08-9886-4ed7-82d8-4ed1e4b82389)

**Comment by Banuk1616** [comment]
@Chacapamac

Thank you very much for your detailed answer to my problem.
I downloaded the fonts (Open Sans and Ropa Sans) from Google and saved them in the media/templates/site/my-template/fonts folder. The fonts are also loaded and displayed on the website, everything works perfectly. But even though I only load the Open Sans and Ropa Sans fonts locally under typography, a connection is still established to Google and the Arizonia font. I don't use the Arizonia font on this website and don't know why it connects to google and the Arizonia font. It is a mystery to me. I have built some websites with Astroid 2.x and never had this problem. It seems like the Arizonia font is loaded from google somewhere in the Astroid Framework.
There are no extensions installed because I'm still working on the design. So it can't be due to an installed extension.
My problem is not that the locally installed fonts don't work, but that the Arizonia font is loaded from Google, which I don't use and I don't know why.

`<link href="[https://fonts.gstatic.com?03e9dc84add78da0330d2b843742e5e5](view-source:https://fonts.gstatic.com/?03e9dc84add78da0330d2b843742e5e5)" rel="preconnect" />
	<link href="[https://fonts.googleapis.com/css2?family=Arizonia&amp;display=swap](view-source:https://fonts.googleapis.com/css2?family=Arizonia&display=swap)" rel="stylesheet" />`

**Reply by Chacapamac** [reply]
**Something somewhere is calling that font?????**
— Did you look in the template manager calls to Custom Code
— Did you looks in all your used Layouts that a font setting is not calling that font
— Did you check if its loaded on all menus/pages or on a section of the site

Do you use something, like Dreamweaver that can search specific strings inside web files, if not I’m sure you can download free software that can do that and search for “Ariziona”

I will try also to search in the Web Inspector and the website database for the same term, that maybe can give you hints.

**Can you put a copy of the website available online?**

**Comment by Banuk1616** [comment]
@Chacapamac
I am very grateful to you for helping me.

I was able to locate the problem.

Under Logo Options -> Logo Type  I selected “None”. But the Arizonia font was still loaded here by Google, even though I had selected Logo Type None.

This is bad!

If you select Logo Type None, the font should not be loaded because selecting None also hides the font setting.

I think the Astroid Framework needs to be improved here. Maybe it's a bug

I only found this by accident because I was searching the root folder with Notepad++ for Arizonia font.

Maybe Sonny can take a look at it.

I thank them very much for their help.

And please excuse my bad English. I'm trying really hard, but I still need Google Translate.

**Reply by sonvnn** [reply]
Please help me check if you are using a Widget which load Arizonia font. Let's check your widgets in Layout Builder.

**Reply by cesarsupport** [reply]
Thanks to this I managed to solve it. 
I selected logo type text, you can change the font Arizonia and then select logo image type  again.
Solved!!

**Comment by Banuk1616** [comment]
@sonvnn 

I was able to locate and fix my problem. I used “no logo” in the logo setting. This also hides the font settings for the logo. I thought if I selected the setting "no logo" that the font settings for the logo would also be deactivated.
I have now changed the font settings to Local under Logo Settings and the Arizona font is no longer loaded by Google

**Reply by sonvnn** [reply]
Opps! Thank you for let me know. I will check this issue asap. Just busy for Jollyany 3.5.24 release.

Thank you very much.

Best Regards,
Sonny

**Reply by Banuk1616** [reply]
Maybe it would make sense if you select "no logo" so that the font settings for the logo are deactivated at the same time.
Why should the hidden fonts still be loaded when you select "no logo"?
If you know it's like that, then that's fine, but I've stumbled upon it now, even though I've already built a few websites with Astroid 2.xx. But that's just my opinion and I still really like the Astroid Framework.

**Comment by Chacapamac** [comment]
@Banuk1616  Happy you fix it.
@sonvnn — You probably have to verify that on the templates masters to be sure no fonts are call by the “logo” settings.

**Reply by Banuk1616** [reply]
I'm also happy that I was able to locate and fix my problems.
So I'm always learning something new. I really like the Astroid Framework, even if I don't always understand it straight away.

**Comment by ghost** [comment]
I have again the problem that somewhere in the local design of my new website a connection to Google Fonts is established.

 `<link href="https://fonts.googleapis.com/css2?family=Arizonia&amp;display=swap" rel="stylesheet" data-asset-name="astroid.googlefont.Arizonia" />`

I've been searching for quite a while now but I can't find where the Google fonts are integrated.  

The possibilities where the Google fonts can be loaded are also very diverse. Every widget can load Google fonts. It is very inconvenient if all fonts are set to local under the typography options, but Google fonts are still loaded in widgets  

It would be great if you could deactivate the loading of Google fonts in the Astroid plugin.

If you run your website in Germany, there can be big problems with the GDPR if you load Google Fonts from Google.

**Reply by cesarsupport** [reply]
Hello.
I have the same problem, it loads the font Arizonia, although the typography I use in the template is " System | Arial". 
Someone knows how to remove the Arizonia font  ?
Thanks.

**Comment by Chacapamac** [comment]
Maybe if you try to do a search in the database or the web inspector (code) - Maybe you will find what and where it’s loaded

**Comment by boldc** [comment]
Anyone ever find a solution to this? Changine the logo font does not resolve the issue,I can't find a reference to Arizona font anywhere else in the template
